src/contract_wrappers/tokens/__tests__/security_token_wrapper.test.ts

```diff
@@ -0,0 +1,2862 @@
+      // Stub the request
+      when(mockedOwnerMethod.callAsync()).thenResolve(expectedOwnerResult);
+      // Mock web3 wrapper owner
+      when(mockedWrapper.getAvailableAddressesAsync()).thenResolve([expectedOwnerResult]);
+
+      const expectedFrozenResult = false;
+      // Mocked method
+      const mockedFrozenMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.mintingFrozen).thenReturn(instance(mockedFrozenMethod));
+      // Stub the request
+      when(mockedFrozenMethod.callAsync()).thenResolve(expectedFrozenResult);
+
+      // Real call
+      const result = await target.mintMulti(mockedParams);
+
+      // Result expectation
+      expect(result).toBe(expectedResult);
+      // Verifications
+      verify(mockedContract.mintMulti).once();
+      verify(
+        mockedMethod.sendTransactionAsync(
+          mockedParams.investors,
+          mockedParams.values,
+          mockedParams.txData,
+          mockedParams.safetyFactor,
+        ),
+      ).once();
+      verify(mockedContract.owner).once();
+      verify(mockedOwnerMethod.callAsync()).once();
+      verify(mockedContract.mintingFrozen).once();
+    });
+  });
+
+  describe('burnWithData', () => {
+    test('should send the transaction to burnWithData', async () => {
+      // Mocked parameters
+      const mockedParams = {
+        value: new BigNumber(1),
+        data: 'string',
+        txData: {},
+        safetyFactor: 10,
+      };
+      const expectedResult = getMockedPolyResponse();
+      // Mocked method
+      const mockedMethod = mock(MockedSendMethod);
+      // Stub the method
+      when(mockedContract.burnWithData).thenReturn(instance(mockedMethod));
+      // Stub the request
+      when(
+        mockedMethod.sendTransactionAsync(
+          mockedParams.value,
+          stringToBytes32(mockedParams.data),
+          mockedParams.txData,
+          mockedParams.safetyFactor,
+        ),
+      ).thenResolve(expectedResult);
+
+      // Owner Address expected
+      const owner = '0x5555555555555555555555555555555555555555';
+      // Mock web3 wrapper owner
+      when(mockedWrapper.getAvailableAddressesAsync()).thenResolve([owner]);
+
+      const expectedBalanceResult = new BigNumber(100);
+      const params = {
+        owner,
+      };
+      // Mocked method
+      const mockedBalanceMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.balanceOf).thenReturn(instance(mockedBalanceMethod));
+      // Stub the request
+      when(mockedBalanceMethod.callAsync(params.owner)).thenResolve(expectedBalanceResult);
+
+      // Real call
+      const result = await target.burnWithData(mockedParams);
+
+      // Result expectation
+      expect(result).toBe(expectedResult);
+      // Verifications
+      verify(mockedContract.burnWithData).once();
+      verify(
+        mockedMethod.sendTransactionAsync(
+          mockedParams.value,
+          stringToBytes32(mocked
```

Comment:
It should be a single `describe` for `addModule` but with all different `test` sentences.
Example:
```
describe('AddModule', () => {
    test('should send the transaction to addModule for CountTransferManager', async () => {
    });
    test('should send the transaction to addModule for PercentageTransferManager', async () => {
    });
});
```
This way is easier to detect which method is not working properly when a test fails.

Reply:
:sweat: